app/Services/Model/Imp/ScheduleService.php

```diff
@@ -120,6 +120,24 @@ public function publishProposedActivityToSource(
 
                 if (!$selection_plan->isAllowProposedSchedules())
                     throw new ValidationException("selection plan id {$selection_plan->getId()} does not allow proposed schedules");
+
+                if (!is_null($schedule)) {
+                    $prevEvents = $schedule->getProposedPublishedEventsBeforeThan($event->getStartDate(), $event->getLocation());
+
+                    foreach ($prevEvents as $prevEvent) {
+                        if (($event->getStartDate()->getTimestamp() - $prevEvent->getEndDate()->getTimestamp()) / 60 < $event->getTrackTransitionTime())
+                            throw new ValidationException(
+                                "There must be a transition time of at least {$event->getTrackTransitionTime()} minutes between the end of the previous event ({$prevEvent->getId()}) and the start of the current one ({$event->getId()})");
+                    }
+
+                    $nextEvents = $schedule->getProposedPublishedEventsAfterThan($event->getEndDate(), $event->getLocation());
```

Comment:
this is not the asked method
please read the SDS
at publishing time ( on proposed schedule) , we need to get the previous and the next published activity at the same location

https://docs.google.com/document/d/1cHQk0ITDrNICMtjluMBdapaEEMw2odUJeH0FEECgRvU/edit

Reply:
done